source/RoslynAnalyzers/CaseSensitiveEnumParsingAnalyzer.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Octopus.RoslynAnalyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CaseSensitiveEnumParsingAnalyzer : DiagnosticAnalyzer
+    {
+        const string DiagnosticId = "Octopus_CaseSensitiveEnumParsing";
+
+        const string Title = "Enum.Parse is case sensitive";
+
+        const string MessageFormat = "The Enum.Parse call is case sensitive, which is likely not intended. Use an overload that explicitely defines case behaviour, or our `.ToEnum()` extension method.";
+        const string Category = "Octopus";
+
+        const string Description = @"Case sensitive Enum parsing has been the cause of several bugs. This analyzer forces the author to specify the casing behavior explicitely.";
+
+        internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Error,
+            true,
+            Description
+        );
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(CheckForCaseSensitiveEnumParsing, SyntaxKind.InvocationExpression);
+        }
+
+        void CheckForCaseSensitiveEnumParsing(SyntaxNodeAnalysisContext context)
+        {
+            if (context.Node is InvocationExpressionSyntax invocation &&
+                invocation.Expression is MemberAccessExpressionSyntax memberAccess &&
+                memberAccess.Expression is IdentifierNameSyntax identifierName &&
+                identifierName.Identifier.Text == "Enum" &&
```

Comment:
May consider using `nameof` here

Reply:
I think that would make it less readable. We tend to use `nameof` to ensure it gets picked up by a refactor (either by the tooling or a failing compile). `nameof` also makes it easier to find usages. None of that applies here since no-one is going to rename `Enum`